src/open_inwoner/js/components/form/FileInput.js

```diff
@@ -179,26 +180,44 @@ export class FileInput extends Component {
   /**
    * Adds files in dataTransfer to input, only the first item is added if not `[multiple]`.
    * @param {File[]} files
+   * @param {boolean} removeCurrent=false
    */
-  addFiles(files) {
+  addFiles(files, removeCurrent = false) {
     const input = this.getInput()
     const dataTransfer = new DataTransfer()
     // Ensure the previously selected files are added as well
-    const _files = input.multiple
-      ? [...input.files, ...files]
-      : [...input.files, files[0]]
+
+    let _files
+    if (removeCurrent) {
+      _files = input.multiple ? [...files] : [files[0]]
+    } else {
+      _files = input.multiple
+        ? [...input.files, ...files]
+        : [...input.files, files[0]]
+    }
 
     _files.filter((v) => v).forEach((file) => dataTransfer.items.add(file))
     input.files = dataTransfer.files
+    this.files = _files
   }
 
   /**
    * Persists state to DOM.
    * NOTE: We inspect the actual input here to obtain the `FileList` state.
    * NOTE: CHANGE EVENT MAY BE BYPASSED WHEN USING HTMX.
    */
-  render() {
-    const { files } = this.getInput()
+  render(e) {
+    let { files } = this.getInput()
+
+    // Bugfix for https://taiga.maykinmedia.nl/project/open-inwoner/task/1975
+    // Selecting files using the select window triggers a change event and would normally
+    // overwrite the previously selected files, so the previously selected files need to
+    // be stored and joined with the newly uploaded files, to avoid the selection from being overwritten
+    if (e && e.type === 'change') {
+      files = [...(this.files || []), ...files]
+      this.addFiles(files, true)
+    }
```

Comment:
Couldn't really find another way to fix this bug for selecting files using the button/window, because it seems to be default behavior to immediately override the input's files, so the previously selected files need to be stored somewhere.

Reply:
Yes this is an incredibly difficult aspect of the default behaviour for the HTML file-input type 😢